### PR TITLE
chore: Update github actions `uses`.

### PR DESCRIPTION
**Pull request by iphydf**
The old ones are going to stop working soon.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2481)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
   mypy:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (robinlinden) [inline .github/workflows/ci.yml:19]: Not v4?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -34,66 +34,66 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
+        uses: docker/setup-buildx-action@v2
```
Comment (robinlinden) [inline .github/workflows/ci.yml:37]: Not v3?

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -17,11 +17,11 @@ jobs:
     steps:
       - name: Login to DockerHub
         if: ${{ github.event_name == 'push' }}
-        uses: docker/login-action@v1
+        uses: docker/login-action@v2
```
Comment (robinlinden) [inline .github/workflows/docker.yml:20]: v3?

**Review by robinlinden — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (iphydf) [reply .github/workflows/ci.yml]: Done.

File: .github/workflows/docker.yml (hunk 3 above)
Comment (iphydf) [reply .github/workflows/docker.yml]: Done.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (iphydf) [reply .github/workflows/ci.yml]: Done.

**Pull request merged**